analyzer/codechecker_analyzer/analyzers/analyzer_base.py

```diff
@@ -68,6 +70,39 @@ def get_analyzer_mentioned_files(self, output):
         """
         raise NotImplementedError("Subclasses should implement this!")
 
+    @classmethod
+    def get_analyzer_version_string(cls, cfg_handler, environ):
```

Comment:
I think we should have the source code for getting the version only at one place. There is already a parser for clangsa [here](https://github.com/Ericsson/codechecker/blob/84469186bcfdebea6f0e42d7b6fd77dce11413aa/analyzer/codechecker_analyzer/analyzers/clangsa/version.py). Couldn't we use that or create something similar to clang tidy and put this version checking/parsing not in the base class but for the specific analyzers?